src/chessboard/components/Board.tsx

```diff
@@ -86,6 +81,19 @@ export function Board() {
           );
         })}
       </svg>
+      {promotion.isDialogOpen && (
+        <SelectPromotionDialog
+          onChange={(option: any) => {
```

Comment:
can this be typed please, we should try to avoid `any` types

Reply:
can the function be hoisted above the jsx too please, and renamed to handlePromotion as mentioned below